setup.py

```diff
@@ -221,7 +223,25 @@ def get_extensions():
         except ImportError:
             pass
 
-        if is_rocm_pytorch or torch.cuda.is_available() or os.getenv(
+        if os.getenv('MMCV_WITH_DIOPI', '0') == '1':
+            print(f'Compiling {ext_name} with CPU and DIPU')
+            define_macros += [('MMCV_WITH_DIOPI', None)]
+            define_macros += [('DIOPI_ATTR_WEAK', None)]
```

Comment:
What is this macro for?

Reply:
for defining  ```__attribute__((weak))``` ref: https://github.com/DeepLink-org/DIOPI/blob/main/DIOPI-PROTO/include/diopi/diopirt.h#LL16C18-L16C40